src/ledc.rs

```diff
@@ -236,19 +221,50 @@ impl<'d> LedcDriver<'d> {
             }
         }
 
-        // SAFETY: As long as we have borrowed the timer, we are safe to use
-        // it.
-        esp!(unsafe { ledc_channel_config(&channel_config) })?;
-
-        Ok(LedcDriver {
-            duty,
-            hpoint,
+        let mut driver = LedcDriver {
+            duty: 0,
+            hpoint: 0,
             speed_mode: timer_driver.borrow().speed_mode,
             max_duty: timer_driver.borrow().max_duty,
             timer: timer_driver.borrow().timer() as _,
             channel: C::channel() as _,
+            pin: 0,
             _p: PhantomData,
-        })
+        };
+
+        driver.set_pin(pin)?;
+
+        driver.apply_config()?;
+
+        Ok(driver)
+    }
+
+    pub fn apply_config(&mut self) -> Result<(), EspError> {
+        let channel_config = ledc_channel_config_t {
+            speed_mode: self.speed_mode.into(),
+            channel: self.channel as u32,
+            timer_sel: self.timer as u32,
+            intr_type: ledc_intr_type_t_LEDC_INTR_DISABLE,
+            gpio_num: self.pin,
+            duty: self.duty,
+            hpoint: self.hpoint as _,
+            ..Default::default()
+        };
+
+        // SAFETY: As long as we have borrowed the timer, we are safe to use
+        // it.
+        esp!(unsafe { ledc_channel_config(&channel_config) })?;
```

Comment:
When a channel is activity running and we switch the config out, can we do this even if the channel is running? Should it be stoped / started again, what does happen if we don't do it? Does it just work without site-effects?

Reply:
from my limited tests, it works as expected without adverse effects

it doubles the output from the old gpio to the new one, the old one has to be manually reset, I see no side effects with doing this switcheroo and changing the duty cycle

Reply:
Can you clarify what you mean by "doubles the output". I expect when changing it while the pwm is running that one pin stops and another starts but not both output at the same time? Can you show a short codesnippet what you mean by manually reset? Did you gave an owned value to the set_pin method or a mut ref?

Reply:
they indeed both output at the same time

this is an example

```rust
    fn configure_direction(&mut self, forward: bool) {
        unsafe {
            if forward {
                gpio_set_direction(self.gpio_backward, gpio_mode_t_GPIO_MODE_OUTPUT);
                gpio_set_level(self.gpio_backward, 0);
                self.driver
                    .config_with_pin(AnyOutputPin::new(self.gpio_forward))
                    .unwrap();
            } else {
                gpio_set_direction(self.gpio_forward, gpio_mode_t_GPIO_MODE_OUTPUT);
                gpio_set_level(self.gpio_forward, 0);
                self.driver
                    .config_with_pin(AnyOutputPin::new(self.gpio_backward))
                    .unwrap();
            }
        }

        self.forward = forward;
    }
    
 ```
 
 it is used to drive a motor forwards or backwards, if you don't do the gpio_set_*, the signal is copied on both pins

Reply:
with the now merged modification, it works perfectly

Reply:
If more people find it useful, maybe one of you would be interested in extending the api in a way to allow directly working with our provided gpio peripherals. There is no strict need for example to use the PinDriver in an internal implementation if constructing / deconstructing here becomes a problem. Though it would be still double with for example an itterator based approach or some sort of closure based approach.